### DISCUSSION TITLE
HOW TO PARSE THIS DATA TO VOLDEMORT/MONGO/PGSQL?????

### DISCUSSION THREAD

**Discussion by NSH531** [PowerShell/PowerShell#17908] [Q&A]
data:

`"@{device_id=e76098504fdf4e57afd8d43cf45d88c5; cid=c75e5cdf759c449c9790798e8d089a27; agent_load_flags=1; agent_local_time=2022-08-16T14:27:40.239Z; agent_version=6.42.15610.0; bios_manufacturer=innotek GmbH; bios_version=VirtualBox; build_number=19043; config_id_base=65994753; config_id_build=15610; config_id_platform=3; cpu_signature=526060; detection_suppression_status=unsuppressed; external_ip=82.166.184.148; mac_address=08-00-27-de-7b-66; hostname=DANIELTESTS; host_hidden_status=visible; first_seen=2021-10-17T10:05:38Z; last_seen=2022-08-16T11:29:22Z; local_ip=10.0.2.15; major_version=10; minor_version=0; os_version=Windows 10; os_build=19043; platform_id=0; platform_name=Windows; policies=System.Object[]; reduced_functionality_mode=no; device_policies=; groups=System.Object[]; group_hash=bdb572eb0f11b1cdfeeb4dfba302c641d10bc97fd289892ebf060fd5c9bd8f73; product_type=1; product_type_desc=Workstation; provision_status=Provisioned; serial_number=0; service_pack_major=0; service_pack_minor=0; pointer_size=8; status=normal; system_manufacturer=innotek GmbH; system_product_name=VirtualBox; tags=System.Object[]; modified_timestamp=2022-08-16T11:32:26Z; meta=; kernel_version=10.0.19043.1889}"`

**Comment by FireInWinter** [comment]
I'm not sure what you are looking for.  Are you looking to create an insert statement from this data?  Or do you want to convert it to a PS Object, where you can then do other operations for it?    Also, where did the data come from?  It looks like it was originally a PS Object based on the "System.Object[]" items in it.  If so, can you get the data from there as it is easier to work with than text data?

But regardless, here's a take at converting it to an object, then you can go from there.
```
$x = "@{device_id=e76098504fdf4e57afd8d43cf45d88c5; cid=c75e5cdf759c449c9790798e8d089a27; agent_load_flags=1; agent_local_time=2022-08-16T14:27:40.239Z; agent_version=6.42.15610.0; bios_manufacturer=innotek GmbH; bios_version=VirtualBox; build_number=19043; config_id_base=65994753; config_id_build=15610; config_id_platform=3; cpu_signature=526060; detection_suppression_status=unsuppressed; external_ip=82.166.184.148; mac_address=08-00-27-de-7b-66; hostname=DANIELTESTS; host_hidden_status=visible; first_seen=2021-10-17T10:05:38Z; last_seen=2022-08-16T11:29:22Z; local_ip=10.0.2.15; major_version=10; minor_version=0; os_version=Windows 10; os_build=19043; platform_id=0; platform_name=Windows; policies=System.Object[]; reduced_functionality_mode=no; device_policies=; groups=System.Object[]; group_hash=bdb572eb0f11b1cdfeeb4dfba302c641d10bc97fd289892ebf060fd5c9bd8f73; product_type=1; product_type_desc=Workstation; provision_status=Provisioned; serial_number=0; service_pack_major=0; service_pack_minor=0; pointer_size=8; status=normal; system_manufacturer=innotek GmbH; system_product_name=VirtualBox; tags=System.Object[]; modified_timestamp=2022-08-16T11:32:26Z; meta=; kernel_version=10.0.19043.1889}"

$hash = @{}; 
$x.trim('}').trim('@').trim('{') -split ';' | % { $_ -match '^(?<setting>[^=]*)=(?<value>.*)$' } | % { $hash.Add($matches.setting.trim(), $matches.value.trim()) } ; 
$Object = [pscustomobject]$hash
```
